agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -457,13 +458,68 @@ bool slave_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
     auto rx_mid = cmdu_rx.getMessageId();
 
     switch (cmdu_message_type) {
-    /* For Example:
+    case ieee1905_1::eMessageType::CHANNEL_PREFERENCE_QUERY_MESSAGE: {
 
-        case ieee1905_1::eMessageType::TOPOLOGY_DISCOVERY_MESSAGE :
-        {
-            break;
+        LOG(INFO) << "received CHANNEL_PREFERENCE_QUERY_MESSAGE";
+
+        // build channel preference report
+        auto cmdu_tx_header =
+            cmdu_tx.create(rx_mid, ieee1905_1::eMessageType::CHANNEL_PREFERENCE_REPORT_MESSAGE);
```

Comment:
don't call it rx_mid, what's wrong with mid?

Reply:
fixed